File: server/memcached/src/main/scala/org/infinispan/server/memcached/MemcachedOperation.scala
package org.infinispan.server.memcached

/**
 * Memcached operations.
 *
 * @author Galder Zamarreño
 * @since 4.1
 */
object MemcachedOperation extends Enumeration {
   type MemcachedOperation = Value
   val PutRequest, PutIfAbsentRequest, ReplaceRequest, ReplaceIfUnmodifiedRequest = Value
   val GetRequest, GetWithVersionRequest = Value
   val RemoveRequest, StatsRequest = Value
   val AppendRequest, PrependRequest = Value
   val IncrementRequest, DecrementRequest = Value
   val FlushAllRequest, VersionRequest = Value
   val VerbosityRequest, QuitRequest = Value
}

File: server/memcached/src/main/java/org/infinispan/server/memcached/MemcachedOperation.java
package org.infinispan.server.memcached;

/**
 * Memcached operations.
 *
 * @author Galder Zamarreño
 * @since 4.1
 * @deprecated since 10.1. Will be removed unless a binary protocol encoder/decoder is implemented.
 */
@Deprecated
public enum MemcachedOperation {
   PutRequest,
   TouchRequest,
   PutIfAbsentRequest,
   ReplaceRequest,
   ReplaceIfUnmodifiedRequest,
   GetRequest,
   GetWithVersionRequest,
   RemoveRequest,
   StatsRequest,
   AppendRequest,
   PrependRequest,
   IncrementRequest,
   DecrementRequest,
   FlushAllRequest,
   VersionRequest,
   VerbosityRequest,
   QuitRequest
   ;
}
